test/relayHub/RelayHub.test.ts

```diff
@@ -0,0 +1,1199 @@
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('2'),
+                    from: relayOwner
+                });
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.registerRelayServer(url, {
+                    from: relayManager
+                });
+
+                const smartWalletTemplate: SmartWalletInstance =
+                    await SmartWallet.new();
+                const senderAccount = web3.eth.accounts.create();
+
+                smartWalletTemplate.initialize(
+                    senderAccount.address,
+                    token.address,
+                    relayWorker,
+                    '0',
+                    '400000'
+                );
+                factory = await createSmartWalletFactory(smartWalletTemplate);
+
+                const calculatedAddr = await factory.getSmartWalletAddress(
+                    gaslessAccount.address,
+                    constants.ZERO_ADDRESS,
+                    relayRequestMisbehavingVerifier.request.index
+                );
+                await token.mint('1', calculatedAddr);
+
+                await assert.isRejected(
+                    relayHubInstance.deployCall(
+                        relayRequestMisbehavingVerifier,
+                        signatureWithMisbehavingVerifier,
+                        { from: relayWorker, gas, gasPrice }
+                    ),
+                    'Unable to initialize SW',
+                    'SW was deployed and initialized'
+                );
+            });
+
+            it('Should faild a deployRequest if Manager is unstaked', async () => {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('1'),
+                    from: relayOwner
+                });
+
+                let relayWorkersBefore = await relayHubInstance.workerCount(
+                    relayManager
+                );
+                assert.equal(
+                    relayWorkersBefore.toNumber(),
+                    0,
+                    `Initial workers must be zero but was ${relayWorkersBefore.toNumber()}`
+                );
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.workerToManager(relayWorker);
+
+                await relayHubInstance.unlockStake(relayManager, {
+                    from: relayOwner
+                });
+
+                //Verifying stake is now unlocked
+                let stakeInfo = await relayHubInstance.getStakeInfo(
+                    relayManager
+                );
+
+                //Moving blocks to be able to unstake
+                await evmMineMany(Number(stakeInfo.unstakeDelay));
```

Comment:
Do you mean we shouldn't wait for the blocks to be mined? If so I don't think we could avoid this. Before allowing the owner to retrieve the stake, they need to wait for a certain number of blocks to be mined, so here we are simulating exactly this.
Please have a look at the [stakeForAddress function](https://github.com/rsksmart/rif-relay-contracts/blob/master/contracts/RelayHub.sol#L312).

Reply:
What I was referring to, its instead of using an async function we could just execute the evmMineMany and keep with the flow. I have seen some tests that do that.

Reply:
Talked with Antonio about this change